src/components/Personalization/dom-actions/dom/getNonce.js

```diff
@@ -0,0 +1,23 @@
+/*
+Copyright 2019 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+
+import { selectNodes } from "../../../../utils/dom";
+
+/**
+ * Returns the nonce if available.
+ * @param {Node} [context=document] defaults to document
+ * @returns {(String|undefined)} the nonce or undefined if not available
+ */
+export default (context = document) => {
+  const n = selectNodes("[nonce]", context);
+  return n && n[0] && (n[0].nonce || n[0].getAttribute("nonce"));
```

Comment:
I read about some backward compatibility issues with the .nonce vs getAttribute (that's why I have both).
I think I can cache the nonce once received (thought about adding that). 
I switched to using the selectNodes just to use our existing tools but I can probably skip it and avoid the array.

Reply:
> I read about some backward compatibility issues with the .nonce vs getAttribute (that's why I have both).

Can we get more information and make sure it's actually necessary. I hate having code we don't understand the need for. If it's actually necessary, please add a comment for the specifics so some poor developer (like me) doesn't come along and remove it later.

Reply:
Yea, so where I read it was  here:
https://developer.mozilla.org/en-US/docs/Web/API/HTMLOrForeignElement/nonce
It's not clear what browsers and where the cut-off for accessing nonce this way was. Given that modern browsers supports the IDL property we could remove getAttribute and add it back if we get any complaints.

Reply:
Thanks @liljenback. I tracked the origin of this down to https://github.com/whatwg/html/issues/2369#issuecomment-280853946. I think it would be good to add a comment referencing this URL, so that some developer down the road doesn't change it without knowing the context of why it's there in the first place.

Reply:
Just to record something you and I talked about yesterday, I think we need to use the selector `script[nonce]` instead of just `[nonce]` because I believe a CSP can define a different nonce for scripts than for styles. We need to make sure we're using the nonce for scripts.

Reply:
I think I'd rather not distinguish between script and style nonces unless it becomes a requirement. I don't see the value of having them be different and if they don't have to it adds an unnecessary complexity.